src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailTaskHistory.jsx

```diff
@@ -1,41 +1,149 @@
+   */
+  const renderTable = () => (
+    <div className="pb-3">
+      <DataTable
+        itemCount={emailTaskHistoryData.length}
+        columns={[
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskType)}`,
+            accessor: 'task_type',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskInputs)}`,
+            accessor: 'task_input',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskId)}`,
+            accessor: 'task_id',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskRequester)}`,
+            accessor: 'requester',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskSubmittedDate)}`,
+            accessor: 'created',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskDuration)}`,
+            accessor: 'duration_sec',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskState)}`,
+            accessor: 'task_state',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskStatus)}`,
+            accessor: 'status',
+          },
+          {
+            Header: `${intl.formatMessage(messages.taskHistoryTableColumnHeaderTaskProgress)}`,
+            accessor: 'task_message',
+          },
+        ]}
+        data={emailTaskHistoryData}
+      />
+    </div>
+  );
+
+  /**
+   * Function that determines which subredender function to call when displaying the Email Task History table.
+   */
+  const renderTaskHistoryTable = () => {
+    if (errorRetrievingData) {
+      return renderError();
+    }
+
+    if (!emailTaskHistoryData.length) {
+      return renderEmpty();
+    }
+
+    return renderTable();
+  };
 
   return (
     <div>
       <div>
         <p>
-          <FormattedMessage
-            id="bulk.email.task.history.section.heading"
-            defaultMessage="To see the status for all email tasks submitted for this course, click this button:"
-            description="Instructions for course staff and admins to view historical bulk course email task data"
-          />
+          {intl.formatMessage(messages.emailTaskHistoryTableSectionButtonHeader)}
         </p>
-        <button type="button" className="btn btn-outline-primary mb-2">
-          <FormattedMessage
-            id="bulk.email.view.task.history.button"
-            defaultMessage="Show Task Email History"
-            description="Button that displays a table with historical bulk email task data for a course-run"
-          />
-        </button>
+        <Button variant="outline-primary" className="btn btn-outline-primary mb-2" onClick={async () => { await fetchEmailTaskHistoryData(); }}>
```

Comment:
I know I didn't mention this on the last PR, but it would be great if we could give feedback to the user in the form of a loading state. Either by using `StatefulButton` from paragon or just a general "disable button and show spinner" while waiting for the api to respond. Since the current version doesn't do this, I'll leave it up to you if you find it necessary (or as a fast-follow improvement)

https://paragon-edx.netlify.app/components/statefulbutton/

Reply:
Oh, this is a good idea and I'll take a look at the documentation! I think this would be a good addition overall.